testdata/generated/search-all.json

```diff
@@ -354,6 +354,37 @@
       "web"
     ]
   },
+  {
+    "name": "sql_input",
+    "title": "SQL Input",
+    "version": "1.0.1",
+    "release": "ga",
+    "description": "Execute custom queries against an SQL database and store the results in Elasticsearch.",
+    "type": "input",
```

Comment:
What are old versions of Kibana do with packages that are not integrations?

Should we filter by `type: integration` when no `type` is passed? and maybe have a `type: all` to list all packages.

Or maybe this is a point where we should start versioning the API?

Reply:
> and maybe have a type: all to list all packages.

I wouldn't like to design an API that won't be used by Fleet at all (KISS).

> Should we filter by type: integration when no type is passed? 

That would be counter-intuitive, you want to search for packages, but for some historical reason, the /search filters some of them. I would only go with that option to prevent any problems on the Fleet side.

> Or maybe this is a point where we should start versioning the API?

It's always a good practice to version it, but not sure if this is a "dead end" to introduce versioning or if we just enable another GET parameter.

Reply:
Well, then going back to the first question, what are current/old versions of Kibana do with packages that are not integrations?

Reply:
Passing the question to @joshdover.

Reply:
This seems like something that could easily be tested manually without code-level knowledge of Fleet. That would be how I test this personally.

Reply:
Ok, I started the 7.16.3 stack and was able to list the `sql_input` package (`type: input`), which is not good. On the other hand if we use Kibana constraint to cut old Kibana versions that should do the trick.

<img width="1350" alt="Zrzut ekranu 2022-05-26 o 13 46 20" src="https://user-images.githubusercontent.com/14044910/170482216-01d75040-6940-4536-a7c7-099dc961ee5c.png">

WDYT @joshdover? I can go any direction, either adjust API or adjust Kibana constraints.

Reply:
> On the other hand if we use Kibana constraint to cut old Kibana versions that should do the trick.

Agree that this can be a good solution. We should probably add a check for this in the package-spec, and we would need to wait to know the version of Kibana that is going to support input packages for this.

Reply:
Sounds like a good idea, but, to be honest, I wouldn't like to block this PR until that happens.

Reply:
Ah yeah, I didn't want to mean that :slightly_smiling_face: But this may delay declaring input package as GA in the spec.

Reply:
@mtojek I would say that each new input package should start with a kibana version constraints higher or equals to 8.3 which should then do the trick isn't it?

Reply:
Yes, Julien, that's the idea. Just waiting for the confirmation/approval from @joshdover.

Reply:
Kibana version constraint enforced by package-spec SGTM, agreed it can be done as a follow up.